src/components/todoview/DoLaterTodoCard.js

```diff
@@ -1,17 +1,38 @@
+import { getAuth } from "@firebase/auth";
 import Image from "next/image";
 import { useAppContext } from "../../context/AppContext";
 import { addToComplete, addToPending, removeTodo } from "../../utils/utils";
 
 function DoLaterTodoCard({ todoItem }) {
-  const { todoDispatch } = useAppContext();
+  const { todoDispatch, db } = useAppContext();
+  const auth = getAuth();
+  let displayName = "";
+  if (auth.currentUser !== null) {
+    displayName = auth.currentUser.displayName;
+  }
 
-  function handleAction(action) {
+  async function handleAction(action) {
```

Comment:
Its good that you tried having a single function and using if else executed required logic. But it would be better and easier to understand to have individual function for each of those buttons